tensorflow/java/src/test/java/org/tensorflow/ShapeTest.java

```diff
@@ -77,4 +80,36 @@ public void nodesInAGraph() {
       assertEquals(5, n.shape().size(1));
     }
   }
+
+  @Test
+  public void equalsWorksCorrectly() {
+    Shape s1 = Shape.make(-1, 2, 3);
+    Shape s2 = s1;
+    Shape s3 = Shape.make(-1, 2, 3);
+    Shape s4 = Shape.make(-1, 2, 4);
+    Shape s5 = Shape.make(-1, 2, 3, 4);
+    Shape s6 = Shape.make(-1, 2);
+    Object o = new Object();
+
+    assertEquals(s1, s2);
```

Comment:
Optional: The tests might be easier to follow if we don't have to refer back to what `s1`, `s3` etc. mean. Perhaps inline the `Shape` objects? So something like:

```java
assertEquals(Shape.make(-1, 2, 3), Shape.make(-1, 2, 3));
assertNotEquals(Shape.make(1, 2), new Object());
assertNotEquals(Shape.make(-1, 2, 3), Shape.make(-1, 2, 4));
assertEquals(Shape.scalar(), Shape.scalar());
```

Reply:
Yes, good comment.  I'll make some changes.  Same with your comment below.  :-)